### PR TITLE
fix(1010): [3] Fix the error from forked repository

### PR DESCRIPTION
**Pull request by sakka2**
## Context
Now, job error happens in PRs from forked repository, because `sd-scm-github` step tries to `git reset` with the commit SHA of forked repository which dose not exist in original one.
This is because there is no `config.prRef` value in argument below.
https://github.com/screwdriver-cd/scm-github/blob/master/index.js#L316

## Objective
In chainPR workflow, we should pass the `prRef` value to the buildFactory.create() method.
To achieve this, 
1. We need to set `prRef` into the events model  when event is created.
2. Then, we can get `prRef` from events model like `event.pr.ref`;

## References
Issue:
https://github.com/screwdriver-cd/screwdriver/issues/1010#issuecomment-478911280

Other PRs:
[1]: data-schema: https://github.com/screwdriver-cd/data-schema/pull/327
[2]: models: https://github.com/screwdriver-cd/models/pull/359

### CODE REVIEW

File: plugins/builds/index.js [hunk 1]
```diff
@@ -27,30 +27,46 @@ const workflowParser = require('screwdriver-workflow-parser');
...
+        pipelineId
     });
+
+    let prRef = '';
+
+    if (event.type === 'pr') {
+        const scm = buildFactory.scm;
+        const pipeline = await pipelineFactory.get(pipelineId);
+        const token = await pipeline.token;
+        const scmConfig = {
```
Comment (jithin1987) [inline plugins/builds/index.js]: do we have to query `scm` each time ? Can `prRef` be stored in PR event ?

Comment (sakka2) [reply plugins/builds/index.js]: Oh, I think it's better, thanks!

File: plugins/builds/index.js [hunk 2]
```diff
@@ -27,30 +27,46 @@ const workflowParser = require('screwdriver-workflow-parser');
...
-                configPipelineSha: event.configPipelineSha,
-                scmContext,
-                start: start !== false
-            });
-        }
-
-        return null;
+async function createBuild({ jobFactory, buildFactory, eventFactory, pipelineFactory,
+    pipelineId, jobName, username, scmContext, build, start }) {
+    const event = await eventFactory.get(build.eventId);
```
Comment (tk3fftk) [inline plugins/builds/index.js]: Do we need some error handling for these `await` statements?

File: plugins/builds/index.js [hunk 3]
```diff
@@ -27,30 +27,46 @@ const workflowParser = require('screwdriver-workflow-parser');
  * @param  {Boolean}  [config.start]        Whether to start the build or not
  * @return {Promise}
  */
```
Comment (tk3fftk) [inline plugins/builds/index.js]: Need to add `pipelineFactory` to JSDoc

Comment (sakka2) [reply plugins/builds/index.js]: We didn't need pipelineFactory, so I deleted.

File: plugins/builds/index.js (hunk 1 above)
Comment (sakka2) [reply plugins/builds/index.js]: @jithin1987 
I fixed to store the `prRef` into the event model.

Please look these.
https://github.com/screwdriver-cd/data-schema/pull/327
https://github.com/screwdriver-cd/models/pull/359

File: plugins/builds/index.js (hunk 2 above)
Comment (sakka2) [reply plugins/builds/index.js]: May be error will be catched here?
`createBuild` method is called from `triggerNextJobs`, and `triggerNextJobs` is called here.
https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/builds/update.js#L208

File: plugins/builds/index.js [hunk 4]
```diff
@@ -27,30 +27,30 @@ const workflowParser = require('screwdriver-workflow-parser');
...
-
-        return null;
+async function createBuild({ jobFactory, buildFactory, eventFactory, pipelineId,
+    jobName, username, scmContext, build, start }) {
+    const event = await eventFactory.get(build.eventId);
+    const job = await jobFactory.get({
+        name: jobName,
+        pipelineId
     });
+    const prRef = event.pr.ref ? event.pr.ref : '';
```
Comment (jithin1987) [inline plugins/builds/index.js]: is `pr` guaranteed to be an object here ? else we should do a check on `event.pr` as well.

Comment (sakka2) [reply plugins/builds/index.js]: @jithin1987 
yes, `pr` is guaranteed to ben an object here. Please see this link.
https://github.com/screwdriver-cd/models/blob/master/lib/eventFactory.js#L416

**Pull request merged**